src/decide.ts

```diff
@@ -169,7 +175,24 @@ function sizeDoesMatch(resultSize, searchee) {
 	const upperBound = length + fuzzySizeThreshold * length;
 	return resultSize >= lowerBound && resultSize <= upperBound;
 }
+function releaseVersionDoesMatch(
+	searcheeName: string,
+	candidateName: string,
+	matchMode: MatchMode,
+) {
+	const searcheeVersionType = searcheeName.match(RP_REGEX);
+	const candidateVersionType = candidateName.match(RP_REGEX);
+	//sets either match arrtype or type to corresponding VersionType
 
+	if (
+		searcheeVersionType?.[0].toLowerCase() !==
+			candidateVersionType?.[0].toLowerCase() ||
+		searcheeVersionType?.groups?.arrtype
+	) {
+		return matchMode !== MatchMode.SAFE;
```

Comment:
This seems to ignore the check when not using `safe`. Also seems to reject release that maches `arrtype`.

Reply:
if arrtype is true, it means that it is defaulted away from PROPER and REPACK and aggregated into Proper...I see very very few releases that use Proper vs all caps. We could also match if arrtype and candidate is proper or repack instead, but if arrtype exists it necessitate risky - because its obviously a renamed file.

for risky concern, maybe we should add something to the logic on the return ? I only kind of understand what you are meaning...it seems like you think their should be more or a different value returned? Maybe I'm missing something.

Reply:
It seems if matchMode is risky or partial, then this check won't apply at all. Which is fine just clarifying.

I don't see why we should check for arrtype. Since it's case sensitive won't it be rejected anyways?

```ts
export const RP_REGEX =
	/(?:\b|\d)(?<type>(?:REPACK\d?|PROPER\d?|v\d))\b/;
```
https://regex101.com/r/qSkVuj/1

The `\b` covers `\s`. Also can't have both `\b` and `\d` before `vX` so `\b|\d` is required.

Reply:
because arrs label repacks and proper as just proper universally

Reply:
I guess also checking if candidateVersionType exists at all would be smart.

Reply:
```ts
	if (
		searcheeVersionType?.[0].toLowerCase() !==
			candidateVersionType?.[0].toLowerCase() ||
		(searcheeVersionType?.groups?.arrtype && candidateVersionType)
	) {
		return matchMode !== MatchMode.SAFE;
	}
```

Reply:
https://regex101.com/r/6kE85o/1


test cases for all variants